### PR TITLE
Add Seasonal Statistic Example

### PR DESCRIPTION
**Pull request by dallasfoster**
# Earth2Studio Pull Request

## Description
This MR adds an example demonstrating how to define a custom statistic and run a seasonal simulation.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: examples/06_seasonal_statistics.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+
+
+# %%
+# Set Up
+# ------
+# With the statistic workflow defined, we now need to create the indivdual components.
+#
+# We need the following:
+#
+# - Prognostic Model: Use the built in FourCastNet model :py:class:`earth2studio.models.px.FCN`.
```
Comment (NickGeneva) [inline examples/06_seasonal_statistics.py]: - Prognostic Model: Use the built in Pangu Weather 24 hour model :py:class:`earth2studio.models.px.Pangu24`.

File: examples/06_seasonal_statistics.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+# Create the IO handler, store in memory
+io = NetCDF4Backend(
+    file_name="outputs/soi.nc",
+)
+
+# Create the custom statistic
+class SOI:
+    """Custom metric calculation the Southern Oscillation Index.
+
+    SOI = ( standardized_tahiti_slp - standardized_darwin_slp ) / soi_normalization
```
Comment (NickGeneva) [inline examples/06_seasonal_statistics.py:199]: Is there a difference between `slp` and `msl` in this instance?

Comment (dallasfoster) [reply examples/06_seasonal_statistics.py:199]: `slp` refers to sea level pressure, which is only defined at sea level, where `msl` refers to mean sea level pressure, which is interpolated in locations that are not at sea level. In this case, both Darwin and Tahiti are at sea level so `msl` should match `slp`.

File: examples/06_seasonal_statistics.py [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+
+nsteps = 60
+nensemble = 1
+io = run_stats(["2022-01-01"], nsteps, nensemble, model, soi, data, io)
+
+
+# %%
+# Post Processing
+# ---------------
+# The last step is to post process our results. Cartopy is a greate library for plotting
```
Comment (NickGeneva) [inline examples/06_seasonal_statistics.py:347]: Just a line plot here

File: examples/06_seasonal_statistics.py [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+
+        file_path = _download_cached(
+            "http://data.longpaddock.qld.gov.au/SeasonalClimateOutlook/SouthernOscillationIndex/SOIDataFiles/DailySOI1933-1992Base.txt"
+        )
+        ds = pd.read_csv(file_path, sep=r"\s+")
+        dates = pd.date_range("1999-01-01", freq="d", periods=len(ds))
+        ds["date"] = dates
+        ds = ds.set_index("date")
+        ds = ds.drop(["Year", "Day", "SOI"], axis=1)
+        ds = ds.rolling(14, min_periods=1).mean().dropna()
```
Comment (NickGeneva) [inline examples/06_seasonal_statistics.py]: Why rolling window of 14 here?

Comment (dallasfoster) [reply examples/06_seasonal_statistics.py]: The data source has noisy, daily SLP for each location but SOI is typically calculated on monthly climatologies. The window of 14 is a quasi-standard 2 week smoothing filter. I think the exact number itself is debatable and could be either 14 or 28.

**Pull request merged**